PWGDQ/Tasks/tableReader_withAssoc.cxx

```diff
@@ -1438,6 +1453,322 @@ struct AnalysisSameEventPairing {
+            continue;
+          }
+
+          auto t1 = a1.template reducedtrack_as<TTracks>();
+          auto t2 = a2.template reducedtrack_as<TTracks>();
+
+          // Avoid self-pairs
+          if (t1.globalIndex() == t2.globalIndex()) {
+            continue;
+          }
+
+          sign1 = t1.sign();
+          sign2 = t2.sign();
+          // store the ambiguity number of the two dilepton legs in the last 4 digits of the two-track filter
+          if (t1.barrelAmbiguityInBunch() > 1) {
+            twoTrackFilter |= (uint32_t(1) << 28);
+          }
+          if (t2.barrelAmbiguityInBunch() > 1) {
+            twoTrackFilter |= (uint32_t(1) << 29);
+          }
+          if (t1.barrelAmbiguityOutOfBunch() > 1) {
+            twoTrackFilter |= (uint32_t(1) << 30);
+          }
+          if (t2.barrelAmbiguityOutOfBunch() > 1) {
+            twoTrackFilter |= (uint32_t(1) << 31);
+          }
+
+          VarManager::FillPair<TPairType, TTrackFillMap>(t1, t2);
+          if constexpr (TTwoProngFitter) {
+            VarManager::FillPairVertexing<TPairType, TEventFillMap, TTrackFillMap>(event, t1, t2, fConfigPropToPCA);
+          }
+
+          // Fill histograms
+          bool isAmbiInBunch = false;
+          bool isAmbiOutOfBunch = false;
+          for (int icut = 0; icut < fNLegCuts; icut++) {
+            if (twoTrackFilter & (uint32_t(1) << icut)) {
+              isAmbiInBunch = (twoTrackFilter & (uint32_t(1) << 28)) || (twoTrackFilter & (uint32_t(1) << 29));
+              isAmbiOutOfBunch = (twoTrackFilter & (uint32_t(1) << 30)) || (twoTrackFilter & (uint32_t(1) << 31));
+              if (sign1 * sign2 < 0) {
+                fHistMan->FillHistClass(histNames[icut][0].Data(), VarManager::fgValues);
+                if (isAmbiInBunch) {
+                  fHistMan->FillHistClass(histNames[icut][3].Data(), VarManager::fgValues);
+                }
+                if (isAmbiOutOfBunch) {
+                  fHistMan->FillHistClass(histNames[icut][6].Data(), VarManager::fgValues);
+                }
+              } else {
+                if (sign1 > 0) {
+                  fHistMan->FillHistClass(histNames[icut][1].Data(), VarManager::fgValues);
+                  if (isAmbiInBunch) {
+                    fHistMan->FillHistClass(histNames[icut][4].Data(), VarManager::fgValues);
+                  }
+                  if (isAmbiOutOfBunch) {
+                    fHistMan->FillHistClass(histNames[icut][7].Data(), VarManager::fgValues);
+                  }
+                } else {
+                  fHistMan->FillHistClass(histNames[icut][2].Data(), VarManager::fgValues);
+                  if (isAmbiInBunch) {
+                    fHistMan->FillHistClass(histNames[icut][5].Data(), VarManager::fgValues);
+                  }
+                  if (isAmbiOutOfBunch) {
+                    fHistMan->FillHistClass(histNames[icut][8].Data(), VarManager::fgValues);
+                  }
+                }
+              }
```

Comment:
The split of the histogramming for in bunch and out of bunch ambiguity could be useful, but if you didn't wanted it for now, you don't have to add it. In my case, for the same-event pairing, i had to understand a bit some of the effects on jpsi. But anyway, i think it might be useful for you as well. So maybe leave it as it is.